org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/batch/ClasspathJsr199.java

```diff
@@ -156,12 +160,29 @@ public char[][][] findTypeNames(String aQualifiedPackageName, String moduleName)
 	public void initialize() throws IOException {
 		if (this.jrt != null) {
 			this.jrt.initialize();
+		} else if (this.location.isModuleOrientedLocation()) {
+			Iterable<Set<Location>> locationsForModules = this.fileManager.listLocationsForModules(this.location);
+			for (Set<Location> locs: locationsForModules) {
+				for (Location loc : locs) {
+					if (loc instanceof LocationWrapper wrapper) {
+						for (Path locPath : wrapper.getPaths()) {
+							File file = locPath.toFile();
+							IModule mod = ModuleFinder.scanForModule(this, file, null, true, null);
+							if (mod != null) {
+								return;
+							}
+						}
+					}
+				}
+			}
 		}
 	}
 
 	@Override
 	public void acceptModule(IModule mod) {
-		// do nothing
+		if (this.jrt != null)
+			return; // do nothing
+		this.module = mod;
```

Comment:
If we need to handle multiple modules, then we probably need to let `initialize()` remember the module, since only there we know the corresponding LocationWrapper.

Reply:
Obsoleted by creating one ClasspathJsr199 for each module / location.